File: Lecture016 Selection Sort/SelectionSort.java

import java.util.Scanner;
import java.util.Arrays;
public class SelectionSort {
    // Selection Sort
    public static void main(String []args){
        // Creating Scanner Object for taking Inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take SIZE of 1D Array from the USER
        System.out.print(" Enter SIZE of 1D Array: ");
        int n = sc.nextInt();
        // Creating Array of Required SIZE
        int [] arr = new int[n];
        // Fill the Array by taking inputs from the USER
        System.out.print(" Enter " + n + " Elements: ");
        for(int i=0; i<arr.length; i++)
            arr[i] = sc.nextInt();
        // Display the Input Array Before Sorting
        System.out.println(" Array Before Sorting is: " + Arrays.toString(arr));
        // Apply Selection Sort
        selectionSort(arr);
        // Display resultant Array (i.e Sorted Array) After Sorting
        System.out.println(" Array After Sorting is: " + Arrays.toString(arr));
    }
    // Method of Selection Sort
    static void selectionSort(int [] arr){
        for(int i=0; i<arr.length; i++){
            // Find the Maximum Index in the Remaining array and Swap with its correct index
            int last = arr.length - i - 1;
            int maxIndex = getMaxIndex(arr, 0, last);
            swap(arr, maxIndex, last);
        }
    }

    static void swap(int [] arr, int a , int b){
        int temp = arr[a];
        arr[a]   = arr[b];
        arr[b]   = temp;
    }

    static int getMaxIndex(int [] arr, int start, int end){
        int max = start;
        for(int i = start; i <= end; i++){
            if(arr[max] < arr[i])
                max = i;
        }
        return max;
    }
}

File: Lecture016 Selection Sort/selectionSort.cpp
#include<iostream>
using namespace std;
void selectionSort(int arr[], int n)
{   
    for(int i = 0; i < n-1; i++ ) {
        int minIndex = i;
        
        for(int j = i+1; j<n; j++) {
            
            if(arr[j] < arr[minIndex]) 
                minIndex = j;
            
        }
        swap(arr[minIndex], arr[i]);
    }
}
void printArray(int arr[], int size) 
{ 
    int i; 
    for (i=0; i < size; i++) 
        cout << arr[i] << " "; 
    cout << endl; 
} 
int main()
{
 int n ;
 cin >> n;
 int arr[] = {6,2,8,4,7};
 selectionSort(arr, n); 
    cout << "Sorted array: \n"; 
    printArray(arr, n); 
    return 0; 
} 
    
    
    
    
    
    
    
    
    
    
    
